tests/_test_process_executor.py

```diff
@@ -284,6 +287,46 @@ def test_del_shutdown(self):
         for p in processes.values():
             p.join()
 
+    @classmethod
+    def _test_recursive_kill(cls, depth=1):
+        executor = cls.executor_type(
+            max_workers=2, context=cls.context,
+            initializer=_executor_mixin.initializer_event,
+            initargs=(_executor_mixin._test_event,))
+        assert executor.submit(sleep_and_return, 0, 42).result() == 42
+        if depth == 0:
```

Comment:
The variable name makes this test hard to understand: I would have expected depth to increase in the inner/nested calls.